src/dualize.jl

```diff
@@ -71,4 +81,102 @@ function set_dualmodel_sense!(dualmodel::MOI.ModelLike, model::MOI.ModelLike)
         error(sense, " is not supported") # Feasibility should be supported?
     end
     return nothing
+end
+
+"""
+    get_constraints_coefficients(model::MOI.ModelLike)
+
+Get the terms of the a0 vector and the constant b as per 
+http://www.juliaopt.org/MathOptInterface.jl/stable/apimanual/#Advanced-1
+"""
+function get_constraints_coefficients(model::MOI.ModelLike, constr_types::Vector{Tuple{DataType, DataType}})
+    # Empty dictionary to store Ai and bi for each cone
+    dict_coeffs = Dict{Any, Tuple{Vector{Float64}, Float64}}()
+
+    for (F, S) in constr_types
+        num_cons_f_s = MOI.get(model, MOI.NumberOfConstraints{F, S}()) # Number of constraints {F, S}
+        # Fill Ai and bi
+        for con_id = 1:num_cons_f_s
+            fill_constraint_terms(dict_coeffs, model, F, S, con_id)
+        end
+    end
+
+    return dict_coeffs
+end
+
+"""
+    get_objective_coefficients(model::MOI.ModelLike)
+
+Get the terms of the a0 vector and the constant b as per 
+http://www.juliaopt.org/MathOptInterface.jl/stable/apimanual/#Advanced-1
+"""
+function get_objective_coefficients(model::MOI.ModelLike)
+    # Empty vector a0 with the number of variables
+    a0 = zeros(Float64, model.num_variables_created)
+
+    # Fill a0 for each term in the objective function
+    for term in model.objective.terms
```

Comment:
This would give cryptic errors or silently do something wrong if the type of `objective` is not `ScalarAffineFunction`.
I usually prefer having a function extract these information and only defining a method for `ScalarAffineFunction`.
This will
1) give a MethodError if the objective is different and
2) allow anyone with a custom objective to make it work with your code by implementing a new method.

Reply:
I tried to do this in the file `objective_coefficients` :) . What do you think?